website/pages/docs/advanced-topics/docker-cli-docker-plugins.md

```diff
@@ -0,0 +1,112 @@
+# Running Docker Plugins With the CLI Docker Image
+
+CloudQuery CLI uses the [Docker CLI](https://docs.docker.com/engine/reference/commandline/cli/) and [Engine API](https://docs.docker.com/engine/api/) to run Docker plugins.
+When using the CloudQuery CLI Docker image, Docker plugins don't work out of the box, as the Docker CLI and Engine API are not available in the container by default.
+This guide will show you how to run Docker plugins with the CloudQuery CLI Docker image using Docker Compose.
+
+## Prerequisites
+
+- [Docker installed (with Docker Compose)](https://docs.docker.com/get-docker/)
+- [CloudQuery CLI](https://docs.cloudquery.io/docs/quickstart)
+
+## Setup
+
+1. Run `cloudquery login` to authenticate with the CloudQuery registry.
+2. Create a file named `spec.yml` with the configuration for the Docker plugin. We will use this spec file to pull the Docker plugin image locally from the private CloudQuery Docker registry. The `airtable` and `postgresql` plugins are used as an example.
+
+```yaml filename="spec.yml"
+kind: source
+spec:
+  name: airtable
+  registry: docker
+  path: docker.cloudquery.io/cloudquery/source-airtable:VERSION_SOURCE_AIRTABLE
+  tables: ["*"]
+  destinations: ["postgresql"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  path: "cloudquery/postgresql"
+  version: "VERSION_DESTINATION_POSTGRESQL"
+```
+3. Run `cloudquery plugin install spec.yml` to pull the Docker plugin image locally.
```

Comment:
Could we just have them run `docker pull <image>`?

Reply:
No that won't work as our registry is private. Also `docker login docker.cloudquery.io` won't work as our registry requires you to pass a team name in a special header https://github.com/cloudquery/plugin-pb-go/blob/78b50f093a96d903467634c085f3be46dcfaf8db/managedplugin/docker.go#L121
There's a way to configure the additional headers but it's a bit complicated

Reply:
It doesn't seem that hard to specify the custom header... https://docs.docker.com/engine/reference/commandline/cli/#custom-http-headers

Reply:
You need to modify your `config.json` and the headers are sent will all requests to all registries I believe

Reply:
We also pass the plugin version since docker doesn't https://github.com/cloudquery/plugin-pb-go/blob/78b50f093a96d903467634c085f3be46dcfaf8db/managedplugin/docker.go#L120 which is dynamic and we can't put it in the `config.json`